connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -700,6 +707,9 @@ public ConnectorType connectorTypeForClass(String connClass) {
      * @return the {@link ConnectorType} of the connector
      */
     public ConnectorType connectorTypeForConfig(Map<String, String> connConfig) {
```

Comment:
After doing some archaeology on the two null checks that are replaced by this one:
* https://github.com/apache/kafka/pull/10822
* https://github.com/apache/kafka/pull/3812

It appears that the null check is required when the method uses information from both the status backing store and the config backing store. In some situations, the config can be null while there are statuses for the connector and/or tasks:
* After a connector is created, a worker reads the new statuses without reading the new configs.
* After a connector is deleted, a worker reads the (absence of) new configs without reading the cleared statuses.

I think these are relatively narrow windows of opportunity for an NPE, but they still exist. In the same spirit as the current change, it would be better to return UNKNOWN in these situations than throw an NPE.

I'll update the method name and Javadoc.